extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/ExtendedTracer.java

```diff
@@ -5,54 +5,317 @@
+   * SpanBuilder}. The span will be ended at the end of the {@link SpanCallback}.
+   *
+   * <p>If an exception is thrown by the {@link SpanCallback}, the span will be marked as error, and
+   * the exception will be recorded.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param spanCallback the {@link SpanCallback} to call
+   * @param <T> the type of the result
+   * @param <E> the type of the exception
+   * @return the result of the {@link SpanCallback}
+   */
+  public <T, E extends Throwable> T call(SpanBuilder spanBuilder, SpanCallback<T, E> spanCallback)
+      throws E {
+    return call(spanBuilder, spanCallback, ExtendedTracer::setSpanError);
+  }
+
+  /**
+   * Runs the given {@link SpanCallback} inside of the span created by the given {@link
+   * SpanBuilder}. The span will be ended at the end of the {@link SpanCallback}.
+   *
+   * <p>If an exception is thrown by the {@link SpanCallback}, the handleException consumer will be
+   * called, giving you the opportunity to handle the exception and span in a custom way, e.g. not
+   * marking the span as error.
+   *
+   * @param spanBuilder the {@link SpanBuilder} to use
+   * @param spanCallback the {@link SpanCallback} to call
+   * @param handleException the consumer to call when an exception is thrown
+   * @param <T> the type of the result
+   * @param <E> the type of the exception
+   * @return the result of the {@link SpanCallback}
+   */
+  @SuppressWarnings("NullAway")
+  public <T, E extends Throwable> T call(
+      SpanBuilder spanBuilder,
+      SpanCallback<T, E> spanCallback,
+      BiConsumer<Span, Throwable> handleException)
+      throws E {
+    Span span = spanBuilder.startSpan();
+    //noinspection unused
+    try (Scope unused = span.makeCurrent()) {
+      return spanCallback.doInSpan();
     } catch (Throwable e) {
-      span.recordException(e);
+      handleException.accept(span, e);
       throw e;
     } finally {
       span.end();
     }
   }
 
-  /** Call the provided {@link Callable} and wrap with a {@link Span} with the provided name. */
-  public <T> T call(String spanName, Callable<T> callable) throws Exception {
-    Span span = delegate.spanBuilder(spanName).startSpan();
-    try (Scope scope = span.makeCurrent()) {
-      return callable.call();
-    } catch (Throwable e) {
-      span.recordException(e);
-      throw e;
-    } finally {
-      span.end();
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param exception the exception that caused the error
+   */
+  private static void setSpanError(Span span, Throwable exception) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(exception);
+  }
+
+  /**
+   * Injects the current context into a string map, which can then be added to HTTP headers or the
+   * metadata of an event.
+   */
+  public Map<String, String> getTextMapPropagationContext() {
```

Comment:
I don't think this syntactic sugar is worth the API surface area.

Reply:
are you suggesting to remove this feature altogether - or just that the user should call the static method in `Propagation` instead?